xarray/core/indexing.py

```diff
@@ -683,7 +690,7 @@ def __deepcopy__(self, memo):
         return type(self)(self.array)
 
 
-class MemoryCachedArray(ExplicitlyIndexedNDArrayMixin):
+class MemoryCachedArray(_ExplicitlyIndexedNDArrayMixinArray):
```

Comment:
can mypy really not handle that mixin class?

Reply:
The issue is the IndexVariable, ExplicitlyIndexedNDArrayMixin lacks .array which is required for IndexVariable, and therefore we need a new minimal class that are common between the two.